core/src/main/scala/kafka/server/epoch/LeaderEpochFileCache.scala

```diff
@@ -50,7 +49,7 @@ trait LeaderEpochCache {
   */
 class LeaderEpochFileCache(topicPartition: TopicPartition, leo: () => LogOffsetMetadata, checkpoint: LeaderEpochCheckpoint) extends LeaderEpochCache with Logging {
   private val lock = new ReentrantReadWriteLock()
-  private var epochs: ListBuffer[EpochEntry] = lock synchronized { ListBuffer(checkpoint.read(): _*) }
+  private var epochs: ListBuffer[EpochEntry] = inWriteLock(lock) { ListBuffer(checkpoint.read(): _*) }
```

Comment:
Object initialisation doesn't guarantee correct initialisation unless the field is final or volatile. I'll revert this change and add a note to KAFKA-5036 though.